index.js

```diff
@@ -40,27 +47,28 @@ function getSchema (schemaName) {
 }
 
 function processEvent (event, context, callback) {
-  let bibOrItem = null
+  let bibItemOrHolding = null
 
   // Determine whether event has Bibs or Items by checking end of eventSourceARN string:
-  if (/\/Bib/.test(event.Records[0].eventSourceARN)) bibOrItem = 'Bib'
-  if (/\/Item/.test(event.Records[0].eventSourceARN)) bibOrItem = 'Item'
+  if (/\/Bib/.test(event.Records[0].eventSourceARN)) bibItemOrHolding = 'Bib'
+  if (/\/Item/.test(event.Records[0].eventSourceARN)) bibItemOrHolding = 'Item'
+  if (/\/Holding/.test(event.Records[0].eventSourceARN)) bibItemOrHolding = 'Holding'
 
   // Fail if the eventSourceARN didn't tell us what we're handling
-  if (!bibOrItem) throw new Error('Unrecognized eventSourceARN. Aborting. ' + event.Records[0].eventSourceARN)
+  if (!bibItemOrHolding) throw new Error('Unrecognized eventSourceARN. Aborting. ' + event.Records[0].eventSourceARN)
 
-  log.debug('Using schema: ', bibOrItem)
+  log.debug('Using schema: ', bibItemOrHolding)
   // db.connect().then(() => getSchema(bibOrItem)).then((schemaType) => {
-  getSchema(bibOrItem).then((schemaType) => {
+  getSchema(bibItemOrHolding).then((schemaType) => {
     // Get array of decoded records:
     var decoded = event.Records.map((record) => {
       const kinesisData = new Buffer(record.kinesis.data, 'base64')
       return schemaType.fromBuffer(kinesisData)
     })
-    log.debug('Processing ' + bibOrItem + ' records: ', decoded)
+    log.debug('Processing ' + bibItemOrHolding + ' records: ', decoded)
 
     // Invoke appropriate updater:
-    var updater = bibOrItem === 'Bib' ? (new BibsUpdater()) : (new ItemsUpdater())
+    var updater = new updaters[bibItemOrHolding]()
```

Comment:
I'm not 100% sure this syntax will work, and the process of writing unit tests for the handler is proving to be tough. Yay or nay on if it will work?

Reply:
It should work